### PR TITLE
Don't generate individual doc entries for inherited Axes/Axis/Tick methods

### PR DESCRIPTION
**Pull request by anntzer**
These methods directly pass through to the superclass and don't warrant
their own page (having a doc page for Tick.add_callback that just
reproes the docstring of Artist.add_callback doesn't add much, and may
in fact make things more confusing for the reader ("oh, it's separately
documented here, perhaps it does something special?").

Also note that the docs for other Artist subclasses (e.g. Line2D,
AxesImage, etc.) don't bother repeating these either.

attn @tacaswell who added these (together with the non-inherited methods, which should definitely be kept) in https://github.com/matplotlib/matplotlib/pull/7176.

This may also improve a bit the ci build, as this means ~20-25%(!) fewer API pages...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
IIRC, I added these to make sure that we had targets for the auto-reference links sphinx will generate.

**Comment by anntzer** [comment]
Does sphinx not link to the parent class as necessary?  Well, we can always wait for https://github.com/matplotlib/matplotlib/pull/14768 to be merged first and we'll see whether this PR breaks the doc build.

**Comment by anntzer** [comment]
Had to fix just a couple of missing references: Axis refers to its axes property but that's something that numpydoc does automatically and can't be tuned; I also changed the auto-rst table gen to link to the correct parent method which provides the docstring, when it is inherited.

File: lib/matplotlib/artist.py [hunk 1]
```diff
@@ -396,8 +396,8 @@ def _default_contains(self, mouseevent, figure=None):
                 return inside, info
             # subclass-specific implementation follows
 
-        The *figure* kwarg is provided for the implementation of
-        `Figure.contains`.
+        The *canvas* kwarg is provided for the implementation of
```
Comment (QuLogic) [inline lib/matplotlib/artist.py]: But the kwarg is `figure`, so why change?

File: lib/matplotlib/artist.py [hunk 2]
```diff
@@ -1380,24 +1372,20 @@ def pprint_setters(self, prop=None, leadingspace=2):
...
             name = self.aliased_name(prop)
-
             lines.append('%s%s: %s' % (pad, name, accepts))
         return lines
 
     def pprint_setters_rest(self, prop=None, leadingspace=4):
         """
-        If *prop* is *None*, return a list of strings of all settable
-        properties and their valid values.  Format the output for ReST
+        If *prop* is *None*, return a list of ReST-formatted strings of all
```
Comment (QuLogic) [inline lib/matplotlib/artist.py]:
```suggestion
        If *prop* is *None*, return a list of reST-formatted strings of all
```

File: lib/matplotlib/artist.py [hunk 3]
```diff
@@ -1331,7 +1313,17 @@ def get_setters(self):
         Get the attribute strings with setters for object.  e.g., for a line,
         return ``['markerfacecolor', 'linewidth', ....]``.
```
Comment (QuLogic) [inline lib/matplotlib/artist.py]: Might as well fix this:
```suggestion
        Get the attribute strings with setters for object.

        For example, for a line, return ``['markerfacecolor', 'linewidth', ....]``.
```

File: lib/matplotlib/artist.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/artist.py]: probably because I was confused...

File: lib/matplotlib/artist.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/artist.py]: ok

File: lib/matplotlib/artist.py (hunk 3 above)
Comment (anntzer) [reply lib/matplotlib/artist.py]: yup

**Comment by tacaswell** [comment]
I am sold on making those pages more readable (by including less noise).  Slightly concerned about bit rot / overlapping docs if we remove these build products that problem is already.


:+1:, but we need to rebuild the missing references.

**Comment by anntzer** [comment]
rebased

**Comment by QuLogic** [comment]
This looks about ready, but all the doc builds failing is probably not good...

**Comment by anntzer** [comment]
let's see if this fixes it...

**Pull request merged**